ECJ incorrectly allows access to non-static local variable from static context

Environment:
ECJ Version:
```
Eclipse Compiler for Java(TM) v20250403-0101, 3.42.0, Copyright IBM Corp 2000, 2020. All rights reserved.
```
Java Version:

```
java 21.0.7 2025-04-15 LTS
Java(TM) SE Runtime Environment (build 21.0.7+8-LTS-245)
Java HotSpot(TM) 64-Bit Server VM (build 21.0.7+8-LTS-245, mixed mode, sharing)
```
Javac Version:

```
javac 21.0.7
```

Description:
We have identified a test program that ECJ compile successfully but javac cannot compile.
However, the program should be invalid.

``` 
public class Test {
    public static void main(String[] strArr) {
        int NUM = 10;
        class C {
            static {
                int a = NUM;
            }
        }
    }
}
```

When compiling this code with javac of JDK 21, the compiler reports an error:
```
>>>javac Test.java
Test.java:6: error: non-static variable NUM cannot be referenced from a static context
                int a = NUM;
                        ^
1 error
```

However, when compiling the same program using ECJ, the program will compile successfully only with warnings.

```
>>>java -jar ecj-4.36M1.jar --release 21 -d . Test.java
----------
1. WARNING in /home/jireh/ista/CrossFuzz/BokaiReproduce/Test.java (at line 4)
        class C {
              ^
The type C is never used locally
----------
2. WARNING in /home/jireh/ista/CrossFuzz/BokaiReproduce/Test.java (at line 6)
        int a = NUM;
            ^
The value of the local variable a is not used
----------
2 problems (2 warnings)
```

Steps to Reproduce:
```
>>>java -jar ecj-4.36M1.jar --release 21 -d . Test.java
```

Expected behavior: ECJ should reject the test program with a compilation error.
Actual behavior:  ECJ compiles the program without any errors and generates an executable file.

--- a/org.eclipse.jdt.core.tests.compiler/src/org/eclipse/jdt/core/tests/compiler/regression/LocalVariableTest.java
+++ b/org.eclipse.jdt.core.tests.compiler/src/org/eclipse/jdt/core/tests/compiler/regression/LocalVariableTest.java
@@ -23,7 +23,7 @@ import org.eclipse.jdt.internal.compiler.impl.CompilerOptions;
 public class LocalVariableTest extends AbstractRegressionTest {
 
 static {
-//	TESTS_NAMES = new String[] { "testBug537033" };
+	TESTS_NAMES = new String[] { "testStaticInitializerInLocalClassAccessingOuterLocalVariable" };
 }
 public LocalVariableTest(String name) {
 	super(name);
@@ -1136,6 +1136,41 @@ public void testStaticInitializerInLocalClassAccessingOuterLocal() {
 		"Cannot make a static reference to the non-static variable parameter\n" +
 		"----------\n");
 }
+
+public void testStaticInitializerInLocalClassAccessingOuterLocalVariable() {
+	runNegativeTest(
+		new String[] {
+			"X.java",
+			"""
+			public class X {
+			    public static void main(String[] strArr) {
+			        int NUM = 10;
+			        class C {
+			            static {
+			                int a = NUM;
+			            }
+			        }
+			    }
+			}
+			"""
+		},
+		this.complianceLevel < ClassFileConstants.JDK16
+		?
+		"----------\n" +
+		"1. ERROR in X.java (at line 5)\n" +
+		"	static {\n" +
+		"	       ^\n" +
+		"Cannot define static initializer in inner type C\n" +
+		"----------\n"
+		:
+		"----------\n" +
+		"1. ERROR in X.java (at line 6)\n" +
+		"	int a = NUM;\n" +
+		"	        ^^^\n" +
+		"Cannot make a static reference to the non-static variable NUM\n" +
+		"----------\n");
+}
+
 public static Class testClass() {
 	return LocalVariableTest.class;
 }

